### PR TITLE
Strong mode and linter updates

### LINKED ISSUE #181: ddc complains about missing type annotations
Error compiling dartdevc module:vector_math|lib/lib__vector_math.js

[error] Missing parameter type for 'calculated'. (package:vector_math/src/vector_math/error_helpers.dart, line 10, col 22)
[error] Missing parameter type for 'correct'. (package:vector_math/src/vector_math/error_helpers.dart, line 10, col 34)
[error] Missing parameter type for 'calculated'. (package:vector_math/src/vector_math/error_helpers.dart, line 21, col 22)
[error] Missing parameter type for 'correct'. (package:vector_math/src/vector_math/error_helpers.dart, line 21, col 34)
[error] Missing variable type for 'outPx'. (package:vector_math/src/vector_math/frustum.dart, line 200, col 9)
[error] Missing variable type for 'outPy'. (package:vector_math/src/vector_math/frustum.dart, line 200, col 16)
[error] Missing variable type for 'outPz'. (package:vector_math/src/vector_math/frustum.dart, line 200, col 23)
[error] Missing variable type for 'outNx'. (package:vector_math/src/vector_math/frustum.dart, line 200, col 30)
[error] Missing variable type for 'outNy'. (package:vector_math/src/vector_math/frustum.dart, line 200, col 37)
[error] Missing variable type for 'outNz'. (package:vector_math/src/vector_math/frustum.dart, line 200, col 44)
[error] Missing return type for 'setEntry'. (package:vector_math/src/vector_math/matrix2.dart, line 44, col 3)
[error] Missing parameter type for 'other'. (package:vector_math/src/vector_math/matrix2.dart, line 148, col 20)
[error] Missing return type for '*'. (package:vector_math/src/vector_math/matrix2.dart, line 217, col 3)
[error] Missing return type for 'setEntry'. (package:vector_math/src/vector_math/matrix3.dart, line 91, col 3)
[error] Missing parameter type for 'other'. (package:vector_math/src/vector_math/matrix3.dart, line 238, col 20)
[error] Missing return type for '*'. (package:vector_math/src/vector_math/matrix3.dart, line 327, col 3)
[error] Missing return type for 'setEntry'. (package:vector_math/src/vector_math/matrix4.dart, line 162, col 3)
[error] Missing parameter type for 'other'. (package:vector_math/src/vector_math/matrix4.dart, line 516, col 20)
[error] Missing parameter type for 'x'. (package:vector_math/src/vector_math/matrix4.dart, line 656, col 18)
[error] Missing parameter type for 'x'. (package:vector_math/src/vector_math/matrix4.dart, line 694, col 22)
[error] Missing parameter type for 'x'. (package:vector_math/src/vector_math/matrix4.dart, line 846, col 14)
[error] Missing parameter type for 'x'. (package:vector_math/src/vector_math/matrix4.dart, line 880, col 18)
[error] Missing parameter type for 'other'. (package:vector_math/src/vector_math/vector2.dart, line 93, col 20)
[error] Missing variable type for 'sum'. (package:vector_math/src/vector_math/vector2.dart, line 145, col 9)
[error] Missing parameter type for 'other'. (package:vector_math/src/vector_math/vector3.dart, line 101, col 20)
[error] Missing parameter type for 'other'. (package:vector_math/src/vector_math/vector4.dart, line 121, col 20)

Please fix all errors before compiling (warnings are okay).

### PR DESCRIPTION
**Pull request by lexaknyazev**
This PR makes all types sound, enables almost all lints. Should fix #181.

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+analyzer:
+  strong-mode:
+    implicit-dynamic: False
+    implicit-casts: False
+  exclude:
+    - test
+
+linter:
```
Comment (johnmccutchan) [inline analysis_options.yaml]: Where did this list of linters come from? Why does vector_math want any of them? How do we decide when to add / remove from this list? How regularly do we update this list?

File: lib/src/vector_math_64/colors.dart [hunk 2]
```diff
@@ -220,14 +220,14 @@ class Colors {
   /// Convert [rgbColor] from rgb color model to the hue, saturation, and
   /// lightness (HSL) color model and store it in [hslColor].
   static void rgbToHsl(Vector4 rgbColor, Vector4 hslColor) {
-    final max = Math.max(Math.max(rgbColor.r, rgbColor.g), rgbColor.b);
-    final min = Math.min(Math.min(rgbColor.r, rgbColor.g), rgbColor.b);
-    final l = (max + min) / 2.0;
-    var h = 0.0;
-    var s = 0.0;
+    final double max = math.max(math.max(rgbColor.r, rgbColor.g), rgbColor.b);
```
Comment (johnmccutchan) [inline lib/src/vector_math_64/colors.dart]: Be consistent with imports. Some files have dart:math imported as Math and some files have dart:math imported as math.

**Comment by johnmccutchan** [comment]
Mostly this change looks fine. I suggest removing the analysis options part of the change and doing another pass for consistency.

**Comment by lexaknyazev** [comment]
Remaining `as Math` imports were in tests. That is fixed with https://github.com/google/vector_math.dart/pull/182/commits/c4d8f5de556d75829b5101728cc127d9b6a4b8e0. 

I see now that it would be better to remove lints, but maybe keep strong mode options to prevent future updates from accidentally breaking DDC?

**Comment by lexaknyazev** [comment]
@johnmccutchan 
Please, tell if 1670aff1b10227f709be7e336cbd84fb79dedecb looks fine, or something more should be done here.

**Comment by robertmuth** [comment]
I have you get a chance, please update 
https://pub.dartlang.org/packages/vector_math#pub-pkg-tab-versions
as well

**Comment by johnmccutchan** [comment]
@robertmuth v2.0.5 has been published.

**Pull request merged**